endpoints/openrtb2/auction.go

```diff
@@ -1305,6 +1320,20 @@ func (deps *endpointDeps) validateAliasesGVLIDs(aliasesGVLIDs map[string]uint16,
 	return nil
 }
 
+func validateRequestExt(req *openrtb_ext.RequestWrapper) error {
+	reqExt, err := req.GetRequestExt()
+	if err != nil {
+		return err
+	}
+
+	prebid := reqExt.GetPrebid()
+	if prebid != nil && prebid.Cache != nil && (prebid.Cache.Bids == nil && prebid.Cache.VastXML == nil) {
+		return errors.New(`request.ext is invalid: request.ext.prebid.cache requires one of the "bids" or "vastxml" properties`)
+	}
+
+	return nil
+}
```

Comment:
Moved logic out of a marshal override, as we now unmarshal before we run validation with the request wrapper.